pip/qsharp/_qsharp.py

```diff
@@ -243,7 +243,15 @@ def callback(output: Output) -> None:
                 pass
         print(output, flush=True)
 
-    return get_interpreter().interpret(source, callback)
+    telemetry_events.on_eval()
+    start_time = monotonic()
+
+    results = get_interpreter().interpret(source, callback)
```

Comment:
I forget, does this throw if the Q# code fails? Do we want to send the end event then?

Reply:
I'll leave it up to the folks who'll be analyzing the telemetry.